platformio.ini

```diff
@@ -737,6 +737,7 @@ src_filter    = ${common.default_src_filter} +<src/HAL/STM32> +<src/HAL/shared/b
 platform      = ststm32@~6.1
 build_flags   = !python Marlin/src/HAL/STM32F1/build_flags.py
   ${common.build_flags}
+  -DARDUINO_ARCH_STM32
```

Comment:
i dont think its required, may mismatch with the other framework for STM32, we have -DARDUINO_ARCH_STM32F1

Reply:
Both flags seem to be present when currently compiling, but when using `framework-arduinoststm32-maple@~3.10000` only the `ARDUINO_ARCH_STM32F1` is present and the build fails.
Adding `ARDUINO_ARCH_STM32` back fixes the build, but I don't know if there is a better fix.

Reply:
built fine without, here 
alfawise_U30              SUCCESS   00:00:34.986

Reply:
look at the commit story of this pr and the CI tests.

Reply:
k, TMC serial issue, may need a fix in another project

Reply:
I'm not sure this is the only file that needs fixing then. Based on the current code it seems to me that `ARDUINO_ARCH_STM32` should be defined for all STM32 variants but I don't have much experience with the codebase.
It seems strange that this flag is removed by changing the maple framework though?

Reply:
imo it was cleaned... there was STM32GENERIC also... in the past, of maybe its in the other framework

Reply:
could be something to change in SoftwareSerialM project

Reply:
see the output of "git grep -n ARDUINO_ARCH_STM32" no more ref in F1... 

STM32GENERIC is maybe a third one... stm cube maybe, dont remember right

Reply:
https://github.com/danieleff/STM32GENERIC .. on ancestor of the common STM32 :p

Reply:
seems weird this missing beginSerial... all boards use serial

Reply:
This is specific to TMCStepper. Apparently the old framework had a define that is missing now. It had worked for me because the Creality boards use standalone drivers.

I can’t look at it right now, but can help review any changes this evening to make sure we are using the correct defines.

Reply:
This is the output I get with `git grep`, that's not in the STM32 directory:
```
Marlin/src/HAL/AVR/u8g_com_HAL_AVR_sw_spi.cpp:56:#if defined(ARDUINO) && !defined(ARDUINO_ARCH_STM32) && !defined(ARDUINO_ARCH_SAM)
Marlin/src/HAL/platforms.h:40:#elif defined(ARDUINO_ARCH_STM32)
Marlin/src/HAL/shared/servo.h:79:#elif defined(ARDUINO_ARCH_STM32)
Marlin/src/core/serial.h:88:#ifdef ARDUINO_ARCH_STM32
Marlin/src/lcd/dogm/HAL_LCD_com_defines.h:53:  #elif defined(ARDUINO_ARCH_STM32)
Marlin/src/lcd/dogm/u8g_dev_st7920_128x64_HAL.cpp:202:#if NONE(__AVR__, ARDUINO_ARCH_STM32, ARDUINO_ARCH_ESP32) || defined(U8G_HAL_LINKS)
Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp:28:#if !defined(U8G_HAL_LINKS) && ANY(__AVR__, ARDUINO_ARCH_STM32, ARDUINO_ARCH_ESP32)
Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp:56:#elif defined(ARDUINO_ARCH_STM32)
Marlin/src/lcd/dogm/ultralcd_st7920_u8glib_rrd_AVR.cpp:200:#endif // !U8G_HAL_LINKS && (__AVR__ || ARDUINO_ARCH_STM32 || ARDUINO_ARCH_ESP32)
Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/compat.h:267:    #if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC)
Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h:48:#ifdef ARDUINO_ARCH_STM32
```

Reply:
in most cases its 
#if defined(__STM32F1__)
#elif defined(ARDUINO_ARCH_STM32)

Reply:
I see only a few cases like that, but I think this should be fixed in a different PR anyway.

Reply:
Okay, I found that `ARDUINO_ARCH_STM32` is removed from the new PIO build script `platformio-build-stm32f1.py` of `framework-arduinoststm32-maple`. I think it's fine for this PR to add that define because it was there before and cannot build and pass tests without. The define can then be migrated away in a different PR, or what do you think?

Reply:
I agree with adding the define for now. I am less sure about the follow-up PR to remove it. There might be 3rd-party dependencies which require it, but we can discuss that in the other PR.

Reply:
If you look at all the CPP files in `HAL/STM32`, you will see that they all exclude their contents for F1 builds with the following check:

`#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC)`

This  tells me that `ARDUINO_ARCH_STM32` is expected to be set with both toolchains, but STM32F1 should also have `STM32GENERIC`.

Reply:
Oh really, I also wanted to keep it but was under the impression that it wasn't preferable after all these comments. Well certainly makes things easier. 😄

Reply:
Just noting that `STM32GENERIC` is not defined for F1 but for F4/F7.

Reply:
> Just noting that `STM32GENERIC` is not defined for F1 but for F4/F7.

Well, one more think I didn't understand correctly 😁

It seems like these checks are not correct, but I don't see a lot of value in reworking them now, when we should be eliminating the Maple HAL soon anyway. We survived with `ARDUINO_ARCH_STM32` defined for STM32F1 builds until now, so I see no problem in leaving it that way for this PR.